### PR TITLE
Admin Router: Improve the readability of the Makefile, better documentation of the ngindox tool

### PR DESCRIPTION
**Pull request by vespian**
## High Level Description

This PR:
* makes the Makefile targets less verbose and thus more readable
* improves the documentation of ngindox tool

## Related PRs:
AR-NEXT: https://github.com/dcos/dcos/pull/1802

### CODE REVIEW

File: packages/adminrouter/extra/src/Makefile.common [hunk 1]
```diff
@@ -109,21 +111,26 @@ api-docs: clean-devkit-container devkit
...
+ifneq ($(shell git ls-files --exclude-standard --others; echo $$?),0)
+	$(error 'Found untracked and unignored changes. Please run `make api-docs` and commit the result. $(DOCS_TEXT)')
+endif
 	@echo 'No changes found -- Admin Router API docs are up to date.' >&2
 
 .PHONY: lint
 lint: clean-devkit-container devkit
-	#FIXME - split it into two targets
-	docker run \
+	@#FIXME - split it into two targets
```
Comment (adamtheturtle) [inline packages/adminrouter/extra/src/Makefile.common]: ❓ Not new so optional, but it isn't clear what the two targets would do.

File: packages/adminrouter/extra/src/Makefile.common [hunk 2]
```diff
@@ -32,22 +32,23 @@ DEVKIT_DOCKER_OPTS := $(DEVKIT_BASE_DOCKER_OPTS) \
 
 .PHONY: clean-devkit-container
```
Comment (adamtheturtle) [inline packages/adminrouter/extra/src/Makefile.common:33]: From the PR that this was split out of, @vespian said that he had tried all changed targets, and was happy with the shown output. This satisfies me.

File: packages/adminrouter/extra/src/README.md [hunk 3]
```diff
@@ -24,9 +24,26 @@ Admin Router runs on both master and agent nodes, each with different configurat
...
 
-Use `make api-docs` to regenerate the YAML and HTML files.
 
-Use `make check-api-docs` to validate that the YAML and HTML files are up to date.
+## Endpoints documentation
+
+All Admin Router endpoints are documented using the
+[ngindox](https://github.com/karlkfi/ngindox) tool, which uses specially
+formatted comments in order to describe endpoint configurations, and
+automatically parse them into HTML documents. Please check the project
```
Comment (adamtheturtle) [inline packages/adminrouter/extra/src/README.md]: `s/parse/parses`

Comment (adamtheturtle) [reply packages/adminrouter/extra/src/README.md]: `s/project/project's`

File: packages/adminrouter/extra/src/Makefile.common (hunk 1 above)
Comment (vespian) [reply packages/adminrouter/extra/src/Makefile.common]: Uhh, this is a very old comment that should have been removed long time ago - IIRC earlier I was also calling here pylint, but it was too problematic and was removed.

Fixing!

**Comment by vespian** [comment]
@adamtheturtle Done, ready for another pass.

File: packages/adminrouter/extra/src/README.md [hunk 4]
```diff
@@ -24,9 +24,26 @@ Admin Router runs on both master and agent nodes, each with different configurat
...
+Admin Router's CI automatically checks if the endpoint documentation generated
+using `ngindox` and embedded into the repository is up to date. If not, the CI
+job fails and the user needs to regenerate the docs and re-submit the PR.
+
+The check is done by generating the documentation during the build stage.
+If, after the `ngindox` run, `git` detects uncommitted changes, then this means
+that the Admin Router configuration differs from the HTML documents that
+are committed into repository. This is done using `make check-api-docs` target.
+
+In order to regenerate the documentation files, one needs to execute the
```
Comment (adamtheturtle) [inline packages/adminrouter/extra/src/README.md:45]: ❓ Does one "commit into" a repository? I'm not sure. I've seen the phrase "and commit the changes" so perhaps change to that.

**Review by adamtheturtle — APPROVED** [review]
Thanks @vespian .

**Review by mhrabovcin — APPROVED** [review]

File: packages/adminrouter/extra/src/README.md (hunk 4 above)
Comment (vespian) [reply packages/adminrouter/extra/src/README.md:45]: `... HTML documents that are committed into repository...`

https://ludwig.guru/s/committed+to+the+repository

Let's leave it as-is.

Comment (adamtheturtle) [reply packages/adminrouter/extra/src/README.md:45]: That link isn't for "committed into repository" as per the text. In fact Ludwig doesn't have relevant results for that - https://ludwig.guru/s/committed+into+repository.

**Pull request merged**